diff -Naur cc/libcap-2.24/libcap/Makefile BUILD/libcap-2.24/libcap/Makefile
--- cc/libcap-2.24/libcap/Makefile	2014-01-06 00:55:03.000000000 +0000
+++ libcap-2.24/libcap/Makefile	2015-03-19 23:50:22.206620284 +0000
@@ -16,7 +16,7 @@
 # no support).
 ifeq ($(LIBATTR),yes)
 FILES += cap_file
-LDFLAGS += -lattr
+LDFLAGS += -lattr -L $(TARGETPREFIX)/lib
 DEPS = -lattr
 endif
 
@@ -43,7 +43,7 @@
 		$< >$@
 
 _makenames: _makenames.c cap_names.list.h
-	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
+	$(HOST_CC) $(BUILD_CFLAGS) $< -o $@
 
 cap_names.h: _makenames
 	./_makenames > cap_names.h
@@ -78,9 +78,9 @@
 	install -m 0644 $(MINLIBNAME) $(LIBDIR)/$(MINLIBNAME)
 	ln -sf $(MINLIBNAME) $(LIBDIR)/$(MAJLIBNAME)
 	ln -sf $(MAJLIBNAME) $(LIBDIR)/$(LIBNAME)
-ifeq ($(FAKEROOT),)
-	-/sbin/ldconfig
-endif
+#ifeq ($(FAKEROOT),)
+#	-/sbin/ldconfig
+#endif
 	mkdir -p -m 0755 $(PKGCONFIGDIR)
 	install -m 0644 libcap.pc $(PKGCONFIGDIR)/libcap.pc
 
diff -Naur cc/libcap-2.24/Make.Rules BUILD/libcap-2.24/Make.Rules
--- cc/libcap-2.24/Make.Rules	2014-01-06 01:16:21.000000000 +0000
+++ libcap-2.24/Make.Rules	2015-03-19 23:45:39.740143365 +0000
@@ -48,12 +48,13 @@
 KERNEL_HEADERS := $(topdir)/libcap/include/uapi
 IPATH += -fPIC -I$(KERNEL_HEADERS) -I$(topdir)/libcap/include
 
-CC := gcc
+HOST_CC := gcc
++CC := $(CROSS_BASE)/bin/$(TARGET)-gcc
 CFLAGS := -O2 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 BUILD_CC := $(CC)
 BUILD_CFLAGS := $(CFLAGS) $(IPATH)
-AR := ar
-RANLIB := ranlib
+AR := $(TARGET)-ar
+RANLIB := $(TARGET)-ranlib
 DEBUG = -g #-DDEBUG
 WARNINGS=-Wall -Wwrite-strings \
         -Wpointer-arith -Wcast-qual -Wcast-align \
@@ -62,7 +63,9 @@
 LD=$(CC) -Wl,-x -shared
 LDFLAGS := #-g
 
-SYSTEM_HEADERS = /usr/include
+#SYSTEM_HEADERS = /usr/include
+SYSTEM_HEADERS = $(TARGETPREFIX)/include
+
 INCS=$(topdir)/libcap/include/sys/capability.h
 LDFLAGS += -L$(topdir)/libcap
 CFLAGS += -Dlinux $(WARNINGS) $(DEBUG)
diff -Naur cc/libcap-2.24/progs/Makefile BUILD/libcap-2.24/progs/Makefile
--- cc/libcap-2.24/progs/Makefile	2013-12-16 04:46:28.000000000 +0000
+++ libcap-2.24/progs/Makefile	2015-03-19 23:51:41.935625476 +0000
@@ -14,7 +14,7 @@
 ifneq ($(DYNAMIC),yes)
 LDFLAGS += --static
 endif
-LDLIBS += -L../libcap -lcap
+LDLIBS += -L../libcap -lcap -Wl,--rpath -Wl,$(TARGETPREFIX)/lib
 
 all: $(BUILD)
 
