diff --git a/Make.Rules b/Make.Rules
index 8393738..3189659 100644
--- a/Make.Rules
+++ b/Make.Rules
@@ -47,12 +47,13 @@ MINOR=22
 KERNEL_HEADERS := $(topdir)/libcap/include
 IPATH += -fPIC -I$(topdir)/libcap/include -I$(KERNEL_HEADERS)
 
-CC := gcc
+HOST_CC := gcc
+CC := $(CROSS_BASE)/bin/$(TARGET)-gcc
 CFLAGS := -O2 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 BUILD_CC := $(CC)
 BUILD_CFLAGS := $(CFLAGS) $(IPATH)
-AR := ar
-RANLIB := ranlib
+AR := $(TARGET)-ar
+RANLIB := $(TARGET)-ranlib
 DEBUG = -g #-DDEBUG
 WARNINGS=-Wall -Wwrite-strings \
         -Wpointer-arith -Wcast-qual -Wcast-align \
@@ -61,7 +62,9 @@ WARNINGS=-Wall -Wwrite-strings \
 LD=$(CC) -Wl,-x -shared
 LDFLAGS := #-g
 
-SYSTEM_HEADERS = /usr/include
+#SYSTEM_HEADERS = /usr/include
+SYSTEM_HEADERS = $(TARGETPREFIX)/include
+
 INCS=$(topdir)/libcap/include/sys/capability.h
 LDFLAGS += -L$(topdir)/libcap
 CFLAGS += -Dlinux $(WARNINGS) $(DEBUG)
diff --git a/libcap/Makefile b/libcap/Makefile
index 4762c60..53dc9aa 100644
--- a/libcap/Makefile
+++ b/libcap/Makefile
@@ -16,7 +16,7 @@ FILES=cap_alloc cap_proc cap_extint cap_flag cap_text
 # no support).
 ifeq ($(LIBATTR),yes)
 FILES += cap_file
-LDFLAGS += -lattr
+LDFLAGS += -lattr -L$(TARGETPREFIX)/lib
 endif
 
 INCLS=libcap.h cap_names.h $(INCS)
@@ -33,7 +33,7 @@ INCLUDE_GPERF_OUTPUT = -include $(GPERF_OUTPUT)
 endif
 
 _makenames: _makenames.c cap_names.list.h
-	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
+	$(HOST_CC) $(BUILD_CFLAGS) $< -o $@
 
 cap_names.h: _makenames
 	./_makenames > cap_names.h
@@ -68,9 +68,9 @@ install: all
 	install -m 0644 $(MINLIBNAME) $(LIBDIR)/$(MINLIBNAME)
 	ln -sf $(MINLIBNAME) $(LIBDIR)/$(MAJLIBNAME)
 	ln -sf $(MAJLIBNAME) $(LIBDIR)/$(LIBNAME)
-ifeq ($(FAKEROOT),)
-	-/sbin/ldconfig
-endif
+#ifeq ($(FAKEROOT),)
+#	-/sbin/ldconfig
+#endif
 
 clean:
 	$(LOCALCLEAN)
diff --git a/progs/Makefile b/progs/Makefile
index ef51dc6..e5b4790 100644
--- a/progs/Makefile
+++ b/progs/Makefile
@@ -14,7 +14,7 @@ BUILD=$(PROGS)
 ifneq ($(DYNAMIC),yes)
 LDFLAGS += --static
 endif
-LDLIBS += -L../libcap -lcap
+LDLIBS += -L../libcap -lcap -Wl,--rpath -Wl,$(TARGETPREFIX)/lib
 
 all: $(BUILD)
 
