diff -up libcap-2.34/libcap/Makefile.1~ libcap-2.34/libcap/Makefile
--- libcap-2.34/libcap/Makefile.1~	2020-01-04 22:19:40.000000000 +0100
+++ libcap-2.34/libcap/Makefile	2020-05-23 20:01:21.193865749 +0200
@@ -21,6 +21,7 @@ PSXOBJS=$(addsuffix .o, $(PSXFILES))
 MAJLIBNAME=$(LIBNAME).$(VERSION)
 MINLIBNAME=$(MAJLIBNAME).$(MINOR)
 GPERF_OUTPUT = _caps_output.gperf
+CFLAGS += -fPIC
 
 all: $(MINLIBNAME) $(STACAPLIBNAME) libcap.pc libpsx.pc $(STAPSXLIBNAME)
 
@@ -48,7 +49,7 @@ libpsx.pc: libpsx.pc.in
 		$< >$@
 
 _makenames: _makenames.c cap_names.list.h
-	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
+	$(BUILD_CC) $(BUILD_CFLAGS) $(BUILD_CPPFLAGS) $< -o $@
 
 cap_names.h: _makenames
 	./_makenames > cap_names.h
@@ -75,10 +76,10 @@ $(MINLIBNAME): $(CAPOBJS)
 	ln -sf $(MAJLIBNAME) $(LIBNAME)
 
 %.o: %.c $(INCLS)
-	$(CC) $(CFLAGS) $(IPATH) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 cap_text.o: cap_text.c $(USE_GPERF_OUTPUT) $(INCLS)
-	$(CC) $(CFLAGS) $(IPATH) $(INCLUDE_GPERF_OUTPUT) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(INCLUDE_GPERF_OUTPUT) -c $< -o $@
 
 cap_test: cap_test.c libcap.h
 	$(CC) $(CFLAGS) $(IPATH) $< -o $@
diff -up libcap-2.34/Makefile.1~ libcap-2.34/Makefile
--- libcap-2.34/Makefile.1~	2020-01-04 22:19:40.000000000 +0100
+++ libcap-2.34/Makefile	2020-05-23 20:01:21.193865749 +0200
@@ -16,7 +16,6 @@ endif
 ifeq ($(GOLANG),yes)
 	$(MAKE) -C go $@
 endif
-	$(MAKE) -C tests $@
 	$(MAKE) -C progs $@
 	$(MAKE) -C doc $@
 	$(MAKE) -C kdebug $@
diff -up libcap-2.34/Make.Rules.1~ libcap-2.34/Make.Rules
--- libcap-2.34/Make.Rules.1~	2020-05-23 20:01:21.193865749 +0200
+++ libcap-2.34/Make.Rules	2020-05-23 20:15:41.742798449 +0200
@@ -49,34 +49,40 @@ MINOR=34
 KERNEL_HEADERS := $(topdir)/libcap/include/uapi
 IPATH += -fPIC -I$(KERNEL_HEADERS) -I$(topdir)/libcap/include
 
-CC := $(CROSS_COMPILE)gcc
+CC ?= $(CROSS_COMPILE)gcc
 DEFINES := -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS := -O2 $(DEFINES)
-BUILD_CC := $(CC)
-BUILD_CFLAGS := -O2 $(DEFINES) $(IPATH)
-AR := $(CROSS_COMPILE)ar
-RANLIB := $(CROSS_COMPILE)ranlib
+CFLAGS ?= -O2
+CFLAGS += $(DEFINES)
+BUILD_CC ?= $(CC)
+BUILD_CFLAGS ?= -O2
+BUILD_CFLAGS += $(DEFINES) $(IPATH)
+AR ?= $(CROSS_COMPILE)ar
+RANLIB ?= $(CROSS_COMPILE)ranlib
 DEBUG = -g #-DDEBUG
 WARNINGS=-Wall -Wwrite-strings \
         -Wpointer-arith -Wcast-qual -Wcast-align \
         -Wstrict-prototypes -Wmissing-prototypes \
         -Wnested-externs -Winline -Wshadow
 LD=$(CC) -Wl,-x -shared
-LDFLAGS := #-g
+LDFLAGS ?= #-g
 LIBCAPLIB := -L$(topdir)/libcap -lcap
 LIBPSXLIB := -L$(topdir)/libcap -lpsx -lpthread
 
 BUILD_GPERF := $(shell which gperf >/dev/null 2>/dev/null && echo yes)
 
-SYSTEM_HEADERS = /usr/include
+LIBCAP_CPPFLAGS = -I$(KERNEL_HEADERS) -I$(topdir)/libcap/include
+LIBCAP_CPPFLAGS += -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+CPPFLAGS += $(LIBCAP_CPPFLAGS)
+BUILD_CPPFLAGS += $(LIBCAP_CPPFLAGS)
 INCS=$(topdir)/libcap/include/sys/capability.h
 LDFLAGS += -L$(topdir)/libcap
-CFLAGS += -Dlinux $(WARNINGS) $(DEBUG)
+CPPFLAGS += -Dlinux
+CFLAGS += $(WARNINGS) $(DEBUG)
 PAM_CAP := $(shell if [ -f /usr/include/security/pam_modules.h ]; then echo yes ; else echo no ; fi)
 INDENT := $(shell if [ -n "$$(which indent 2>/dev/null)" ]; then echo "| indent -kr" ; fi)
 DYNAMIC := $(shell if [ ! -d "$(topdir)/.git" ]; then echo yes; fi)
 
-GOLANG := $(shell if [ -n "$(shell go version 2>/dev/null)" ]; then echo yes ; else echo no ; fi)
+GOLANG ?= $(shell if [ -n "$(shell go version 2>/dev/null)" ]; then echo yes ; else echo no ; fi)
 ifeq ($(GOLANG),yes)
 GOROOT := $(shell go env GOROOT)
 GOCGO := $(shell if [ "$(shell go env CGO_ENABLED)" = 1 ]; then echo yes ; else echo no ; fi)
diff -up libcap-2.34/pam_cap/Makefile.1~ libcap-2.34/pam_cap/Makefile
--- libcap-2.34/pam_cap/Makefile.1~	2019-12-23 01:12:31.000000000 +0100
+++ libcap-2.34/pam_cap/Makefile	2020-05-23 20:01:21.193865749 +0200
@@ -3,6 +3,8 @@
 topdir=$(shell pwd)/..
 include ../Make.Rules
 
+CFLAGS += -fPIC
+
 all: pam_cap.so
 	$(MAKE) testlink
 
@@ -16,16 +18,16 @@ install: all
 # written (and you know why it fails), email me and explain why. Thanks!
 
 pam_cap.so: pam_cap.o
-	$(LD) -o pam_cap.so $< $(LIBCAPLIB) $(LDFLAGS)
+	$(LD) $(CFLAGS) -o pam_cap.so $< $(LIBCAPLIB) $(LDFLAGS)
 
 pam_cap.o: pam_cap.c
-	$(CC) $(CFLAGS) $(IPATH) -c $< -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 test_pam_cap: test_pam_cap.c pam_cap.c
 	$(CC) $(CFLAGS) $(IPATH) -o $@ test_pam_cap.c $(LIBCAPLIB) $(LDFLAGS) --static
 
 testlink: test.c pam_cap.o
-	$(CC) $(CFLAGS) -o $@ $+ -lpam -ldl $(LIBCAPLIB) $(LDFLAGS)
+	$(CC) $(CFLAGS) $(CPPFLAGS) -o $@ $+ -lpam -ldl $(LIBCAPLIB) $(LDFLAGS)
 
 test: pam_cap.so
 	make testlink
diff -up libcap-2.34/progs/Makefile.1~ libcap-2.34/progs/Makefile
--- libcap-2.34/progs/Makefile.1~	2020-05-23 20:01:21.193865749 +0200
+++ libcap-2.34/progs/Makefile	2020-05-23 20:16:09.625029036 +0200
@@ -23,7 +23,7 @@ $(BUILD): %: %.o $(DEPS)
 	$(CC) $(CFLAGS) -o $@ $< $(LIBCAPLIB) $(LDFLAGS)
 
 %.o: %.c $(INCS)
-	$(CC) $(IPATH) $(CAPSH_SHELL) $(CFLAGS) -c $< -o $@
+	$(CC) $(CPPFLAGS) $(IPATH) $(CAPSH_SHELL) $(CFLAGS) -c $< -o $@
 
 install: all
 	mkdir -p -m 0755 $(FAKEROOT)$(SBINDIR)
diff -up libcap-2.34/tests/Makefile.1~ libcap-2.34/tests/Makefile
--- libcap-2.34/tests/Makefile.1~	2020-05-23 20:01:21.193865749 +0200
+++ libcap-2.34/tests/Makefile	2020-05-23 20:17:22.723523453 +0200
@@ -16,6 +16,8 @@ test: run_psx_test run_libcap_psx_test
 
 sudotest: test run_libcap_launch_test run_libcap_launch_test
 
+CFLAGS += -fPIC
+
 install: all
 
 run_psx_test: psx_test psx_test_wrap
@@ -23,16 +25,16 @@ run_psx_test: psx_test psx_test_wrap
 	./psx_test_wrap
 
 psx_test: psx_test.c $(DEPS)
-	$(CC) $(CFLAGS) $(IPATH) -DNOWRAP $< -o $@ $(LIBPSXLIB)
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(IPATH) -DNOWRAP $< -o $@ $(LIBPSXLIB)
 
 psx_test_wrap: psx_test.c $(DEPS)
-	$(CC) $(CFLAGS) $(IPATH) $< -o $@ $(LIBPSXLIB) -Wl,-wrap,pthread_create
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(IPATH) $< -o $@ $(LIBPSXLIB) -Wl,-wrap,pthread_create
 
 run_libcap_psx_test: libcap_psx_test
 	./libcap_psx_test
 
 libcap_psx_test: libcap_psx_test.c $(DEPS)
-	$(CC) $(CFLAGS) $(IPATH) $< -o $@ $(LIBCAPLIB) $(LIBPSXLIB) -Wl,-wrap,pthread_create --static
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(IPATH) $< -o $@ $(LIBCAPLIB) $(LIBPSXLIB) -Wl,-wrap,pthread_create --static
 
 run_libcap_launch_test: libcap_launch_test libcap_psx_launch_test noop
 	sudo ./libcap_launch_test
