diff --git a/pam_cap/Makefile b/pam_cap/Makefile
index 6f07b6b..d3cd14d 100644
--- a/pam_cap/Makefile
+++ b/pam_cap/Makefile
@@ -7,7 +7,7 @@ include ../Make.Rules
 # that this next line does *not* require -lpam on it.) If you think it
 # does, *verify that it does*, and if you observe that it fails as
 # written (and you know why it fails), email me and explain why. Thanks!
-LDLIBS += -L../libcap -lcap
+LDLIBS += -L../libcap -lcap -lpam
 
 CFLAGS += -fPIC
 
